### PR TITLE
Refactor third-party-trackers.ts

### PR DESCRIPTION
**Pull request by jxxe**
- Clean up `third-party-trackers.ts`
- Use a more realistic site (amazon.com) in `example.js`
- Fix build "[error](https://stackoverflow.com/questions/38209699/error-ts2403-subsequent-variable-declarations-must-have-the-same-type)"

### CODE REVIEW

File: src/third-party-trackers.ts [hunk 1]
```diff
@@ -1,118 +1,78 @@
...
-                if (blockRequests) {
-                    request.abort();
-                    blocked = true;
-                }
+                ],
+                type: 'TrackingRequest',
+                url: request.url()
+            });
+
+            break;
```
Comment (simonft) [inline src/third-party-trackers.ts:67]: Was this previously making multiple eventDataHandler() calls if multiple filters matched?

File: example.js [hunk 2]
```diff
@@ -1,30 +1,21 @@
-const { collector } = require("./build");
-const { join } = require("path");
+const { collector } = require('./build');
+const { join } = require('path');
 
 (async () => {
-  const EMULATE_DEVICE = "iPhone 13 Mini";
+    const URL = 'amazon.com';
```
Comment (simonft) [inline example.js:5]: What's the thought behind this change? I'm guessing you wanted to change it to something that actually returned some results?

File: example.js [hunk 3]
```diff
@@ -1,30 +1,21 @@
...
+    const EMULATE_DEVICE = 'iPhone 13 Mini';
 
-  // Save the results to a folder
-  let OUT_DIR = true;
+    const config = {
+        inUrl: `http://${URL}`,
+        numPages: 3,
+        headless: true,
+        emulateDevice: EMULATE_DEVICE,
+        outDir: join(__dirname, 'demo-dir')
```
Comment (simonft) [inline example.js:13]: Maybe add a comment here along the lines of `// optional, if omitted results will be returned but not saved to disk`, which I'm assuming is the point the original code was trying to get across? Unless that's not actually what happens if you omit it.

**Review by simonft — CHANGES_REQUESTED** [review]

File: example.js (hunk 3 above)
Comment (jxxe) [reply example.js:13]: It returns data either way. If you don't specify the directory, it creates a temporary directory for the logging library to write to, then deletes it at the end. This is something I want to clean up eventually.

```typescript
    if (outDir.includes('bl-tmp')) {
        clearDir(outDir, false);
    }
    return { status: 'success', ...output, reports };
};
```

File: example.js (hunk 2 above)
Comment (simonft) [reply example.js:5]: Can we change this back to example.com for now? I get wanting to change it to something with results but I want to think a bit more about if we want to do that, and if so what it should be. One of the problems with making it amazon.com is some people expect to see a lot more tracking and think something is broken when there's not. See https://themarkup.org/the-breakdown/2020/10/01/i-scanned-my-favorite-social-media-site-on-blacklight-and-it-came-up-pretty-clean-whats-going-on.

Comment (jxxe) [reply example.js]: Amazon actually has a ton of tracking, but I've changed it back. It's not super important.

File: src/third-party-trackers.ts (hunk 1 above)
Comment (jxxe) [reply src/third-party-trackers.ts:67]: No, `PuppeteerBlocker.match()` only returns the first match.

**Review by simonft — APPROVED** [review]
Can you change the commit message to remove the reference to example.js? After that looks good.

**Comment by simonft** [comment]
Also can you squash and merge this, rather than just a regular merge?

**Pull request merged**